lib/pathwayManager.js

```diff
@@ -0,0 +1,321 @@
+import fs from 'fs';
+import { BlobServiceClient } from '@azure/storage-blob';
+import AWS from 'aws-sdk';
+import pkg from '@azure/eventgrid';
+const { EventGridClient } = pkg;
+
+class StorageStrategy {
+  async load() { throw new Error('Not implemented'); }
+  async save(data) { throw new Error('Not implemented'); }
+  watch(callback) { throw new Error('Not implemented'); }
+}
+
+class LocalStorage extends StorageStrategy {
+  constructor(filePath) {
+    super();
+    this.filePath = filePath;
+  }
+
+  async load() {
+    try {
+      const data = await fs.promises.readFile(this.filePath, 'utf8');
+      return JSON.parse(data);
+    } catch (error) {
+      console.error(`Error loading pathways from ${this.filePath}:`, error);
+      return {};
+    }
+  }
+
+  async save(data) {
+    await fs.promises.writeFile(this.filePath, JSON.stringify(data, null, 2));
+  }
+
+  watch(callback) {
+    fs.watch(this.filePath, (eventType) => {
+      if (eventType === 'change') {
+        this.load().then(callback);
+      }
+    });
+  }
+}
+
+class AzureBlobStorage extends StorageStrategy {
+  constructor(connectionString, containerName) {
+    super();
+    this.blobServiceClient = BlobServiceClient.fromConnectionString(connectionString);
+    this.containerClient = this.blobServiceClient.getContainerClient(containerName);
+    this.eventGridClient = new EventGridClient(/* Event Grid credentials */);
+  }
+
+  async load() {
+    try {
+      const blobClient = this.containerClient.getBlobClient('pathways.json');
+      const downloadBlockBlobResponse = await blobClient.download();
+      const data = await streamToString(downloadBlockBlobResponse.readableStreamBody);
+      return JSON.parse(data);
+    } catch (error) {
+      console.error('Error loading pathways from Azure Blob Storage:', error);
+      return {};
+    }
+  }
+
+  async save(data) {
+    try {
+      const blobClient = this.containerClient.getBlobClient('pathways.json');
+      const content = JSON.stringify(data, null, 2);
+      await blobClient.upload(content, content.length);
+    } catch (error) {
+      console.error('Error saving pathways to Azure Blob Storage:', error);
+    }
+  }
+
+  watch(callback) {
+    // Subscribe to Azure Event Grid for blob changes
+    this.eventGridClient.subscribe(
+      /* Event Grid topic endpoint */
+      {
+        filter: {
+          includedEventTypes: ['Microsoft.Storage.BlobCreated', 'Microsoft.Storage.BlobDeleted']
+        }
+      },
+      async (event) => {
+        if (event.data.url.endsWith('pathways.json')) {
```

Comment:
Is the scope of this the whole Azure account?  Like any pathways.json that changes anywhere will trigger this and this gets an event on every single create or delete?  Is there no change?  All changes are delete/creates?  Anyway I'm concerned about the volume of calls here if you hook up to an account with a lot of activity.  Can we include something else in the filter?